source/extensions/filters/http/cache/http_cache_utils.cc

```diff
@@ -88,15 +89,38 @@ void eatDirectiveArgument(absl::string_view& s) {
 // SystemTime::duration::zero(). If parsing overflows, returns
 // SystemTime::duration::max().
 SystemTime::duration eatLeadingDuration(absl::string_view& s) {
-  const absl::string_view::iterator digits_end = c_find_if_not(s, &absl::ascii_isdigit);
-  const size_t digits_length = digits_end - s.begin();
-  if (digits_length == 0) {
-    return SystemTime::duration::zero();
+  absl::optional<uint64_t> n = StringUtil::readAndRemoveLeadingDigits(s);
```

Comment:
I see your point about the differences in eatLeadingDuration and readAndRemoveLeadingDigits. We're better off just leaving EatLeadingDuration as it is.